### PR TITLE
Switch to REST API for bugzilla

### PR DESCRIPTION
**Pull request by aylen384**
Ticket: https://progress.opensuse.org/issues/120276

Not tested E2E as REST is not available on prod instance, yet.

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]
As JSONRPC will be available on the next instance as well and we can't merge this right now to not disrupt execution against the current production instance we should for sure not merge this. However you could test it against the production instance. Not sure why you didn't. Also I understood that we still need a username next to the API key

**Comment by aylen384** [comment]
>  However you could test it against the production instance. Not sure why you didn't. 

Because the production bugzilla doesn't have REST yet.

> Also I understood that we still need a username next to the API key

No we don't.
Actually this is enough:
```
curl https://bugzilla-next.suse.de/rest/bug/1204384?api_key=BLUB
```

And even the API key only seems to be required for internal bugs.


I'm as well for keeping this PR open until I could test the changes against the new bugzilla instance and add fixes if necessary.

File: openqa_bugfetcher/issues/bugzilla_issue.py [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
-            assert req.ok
+            assert req.ok, f"Server Status != 200: {req.text}"
             data = req.json()
             assert data, "Empty JSON Object"
-            if data["error"] and data["error"]["code"] in (100, 101):
-                self.existing = False
-            elif data["error"] and data["error"]["code"] == 102:
-                raise AssertionError("Insufficient permission to access this bug")
+            if data.get("error"):
+                if data["code"] in (100, 101):
```
Comment (perlpunk) [inline openqa_bugfetcher/issues/bugzilla_issue.py:34]: Just checking - did it really change from `data["error"]["code"]` to `data["code"]`?

Comment (aylen384) [reply openqa_bugfetcher/issues/bugzilla_issue.py:34]: yes.

```
curl -s https://bugzilla-next.suse.de/rest/bug/1204384 | json_pp
{
   "code" : 102,
   "documentation" : "https://bugzilla.readthedocs.org/en/5.0/api/",
   "error" : true,
   "message" : "You are not authorized to access bug #1204384. To see this bug, you must first log in to an account with the appropriate permissions."
}
```

**Review by mimi1vx — APPROVED** [review]

**Review by Martchus — APPROVED** [review]
The code itself looks fine. I suppose this still needs testing against a real instance, though.

**Comment by aylen384** [comment]
Merged, as with the new bugzilla update the bugfetcher fails without this patch.

**Pull request merged**